UnboundLib/GameModes/IGameModeHandler.cs

```diff
@@ -27,6 +27,13 @@ public interface IGameModeHandler
 
         void PlayerDied(Player killedPlayer, int playersAlive);
 
+        /// <summary>
+        ///     Should handle cleanup when a player leaves the game mid-match, ideally in such a way that allows the game to continue.
+        ///     The default implementation takes care of reassigning playerIDs and teamIDs to ensure they are sequential, but any extra
+        ///     handling must be done here.
+        /// </summary>
+        void OnPlayerLeftRoom(Photon.Realtime.Player otherPlayer);
```

Comment:
A bit nitpicky: I think the should be `void PlayerLeft(Player player);` to make it more consistent with the other methods. Passing `Photon.Realtime.Player` also smells to me; I don't think gamemodes really need to worry about _photon_ players.

Reply:
I'm just copying the photon logic and naming. Moreover I can't pass the `Player` that left, since RWF (and presumably other gamemodes if someone wanted to) support multiple players per Photon connection. The only other thing I could pass would be the `actorID` of the Photon player that left.